control_panel_api/tests/test_models.py

```diff
@@ -262,6 +260,25 @@ def test_aws_delete(self, mock_detach_bucket_access_from_app_role):
             self.app_1.iam_role_name
         )
 
+    def test_repo_name(self):
+        app = mommy.prepare(
+            'control_panel_api.App',
+            repo_url='https://github.com/org/a_repo_name'
+        )
+        self.assertEqual('a_repo_name', app._repo_name)
+
+        app.repo_url = 'git@github.com:org/repo_2.git'
+        self.assertEqual('repo_2', app._repo_name)
+
+        app.repo_url = 'https://github.com/org/a_repo_name/'
+        self.assertEqual('a_repo_name', app._repo_name)
+
+        app.repo_url = 'http://foo.com'
+        self.assertEqual('foo.com', app._repo_name)
+
+        app.repo_url = 'http://foo.com/'
+        self.assertEqual('foo.com', app._repo_name)
```

Comment:
These test cases should be in a list and tested with something like:
```python
url_test_cases = [
    ('git@github.com:org/a_repo_name', 'a_repo_name'),
    ...
]
for url, expected in url_test_cases:
    app.repo_url = url
    self.assertEqual(expected, app._repo_name)
```

Reply:
👍